File: scripts/freezer.js
const freezer = extendContent(GenericCrafter, "freezer", {
	load() {
		this.super$load();
		this.regions = [];
		this.regions[0] = Core.atlas.find(this.name);
		this.regions[1] = Core.atlas.find(this.name + "-liquid");
		this.regions[2] = Core.atlas.find(this.name + "-top");
	}
});

freezer.buildType = () => extendContent(GenericCrafter.GenericCrafterBuild, freezer, {
	draw() {
	    Draw.rect(freezer.regions[0], this.x, this.y);
	    var liquid = freezer.consumes.get(ConsumeType.liquid).liquid;
	    Drawf.liquid(freezer.regions[1], this.x, this.y, this.liquids.get(liquid) / freezer.liquidCapacity, Liquids.water.color);
	    Draw.rect(freezer.regions[2], this.x, this.y);
	}
});

File: src/revision/world/blocks/production/Freezer.kt
package revision.world.blocks.production

import arc.Core
import arc.graphics.g2d.Draw
import arc.graphics.g2d.TextureRegion
import mindustry.content.Liquids
import mindustry.graphics.Drawf
import mindustry.world.blocks.production.GenericCrafter

open class Freezer(name: String) : GenericCrafter(name) {

    lateinit var liquidRegion: TextureRegion
    lateinit var topRegion: TextureRegion

    override fun load() {
        super.load()
        liquidRegion = Core.atlas.find("$name-liquid")
        topRegion = Core.atlas.find("$name-top")
    }

    override fun icons(): Array<TextureRegion> {
        return arrayOf(region, topRegion)
    }

    inner class FreezerBuild : GenericCrafterBuild() {
        override fun draw() {
            super.draw()
            Drawf.liquid(liquidRegion, x, y, liquids.get(Liquids.water) / liquidCapacity, Liquids.water.color)
            Draw.rect(topRegion, x, y)
        }
    }
}